modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/portlet/action/DeleteCTCommentMVCResourceCommand.java

```diff
@@ -33,13 +43,50 @@ protected void doServeResource(
 			ResourceRequest resourceRequest, ResourceResponse resourceResponse)
 		throws Exception {
 
+		ThemeDisplay themeDisplay = (ThemeDisplay)resourceRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
 		long ctCommentId = ParamUtil.getLong(resourceRequest, "ctCommentId");
 
+		CTComment ctComment = ctCommentLocalService.getCTComment(ctCommentId);
+
+		Role administratorRole = _roleLocalService.fetchRole(
+			themeDisplay.getCompanyId(), RoleConstants.ADMINISTRATOR);
+
+		Role publicationAdministratorRole = _roleLocalService.fetchRole(
+			themeDisplay.getCompanyId(), PublicationRoleConstants.NAME_ADMIN);
+
+		if (((administratorRole == null) &&
+			 !_roleLocalService.hasUserRole(
+				 themeDisplay.getUserId(),
+				 publicationAdministratorRole.getRoleId())) ||
+			((publicationAdministratorRole == null) &&
+			 !_roleLocalService.hasUserRole(
+				 themeDisplay.getUserId(), administratorRole.getRoleId()) &&
```

Comment:
The condition doesn't look right...you're checking the wrong role with the null check

```
((administratorRole == null) &&
			 !_roleLocalService.hasUserRole(
				 themeDisplay.getUserId(),
				 publicationAdministratorRole.getRoleId())
```

Let's extract the admin check to a helper so it's easier to read.

```
private boolean _hasAdminRole() {
    // use permissionChecker to check for the administrator role
    if (permissionChecker.isCompanyAdmin()) {
       return true;
    }

    if (hasPublicationAdminRole) {
        return true;
    }

    return false;
}
```